### PR TITLE
[3.6] stop single host agency from logging too early

### PR DESCRIPTION
**Pull request by kvahed**
https://jenkins01.arangodb.biz/view/PR/job/arangodb-matrix-pr/7767/

### CODE REVIEW

File: arangod/Agency/State.cpp [hunk 1]
```diff
@@ -95,8 +95,16 @@ inline static std::string stringify(index_t index) {
 bool State::persist(index_t index, term_t term, uint64_t millis,
                     arangodb::velocypack::Slice const& entry,
                     std::string const& clientId) const {
-  LOG_TOPIC("b735e", TRACE, Logger::AGENCY) << "persist index=" << index << " term=" << term
-                                   << " entry: " << entry.toJson();
+  LOG_TOPIC("b735e", TRACE, Logger::AGENCY)
+    << "persist index=" << index << " term=" << term << " entry: " << entry.toJson();
+
+#ifdef MAINTAINER_MODE // we should never get here
```
Comment (jsteemann) [inline arangod/Agency/State.cpp]: the ifdef name is `ARANGODB_ENABLE_MAINTAINER_MODE:
```suggestion
#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
```

File: arangod/Agency/Agent.h [hunk 2]
```diff
@@ -468,6 +472,9 @@ class Agent final : public arangodb::Thread, public AgentInterface {
   /// since the epoch of the steady clock.
   std::atomic<int64_t> _leaderSince;
 
+  /// @brief load() has completed
+  std::atomic<bool> _loaded;
```
Comment (jsteemann) [inline arangod/Agency/Agent.h]: `_loaded` is not initialized, and thus has a random value at startup :thinking:

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/8356/

**Comment by KVS85** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/8393/

**Comment by KVS85** [comment]
Tests blue except known false positive on macOS in unreliable catch test.

**Pull request merged**